actions-services.yml

```diff
@@ -0,0 +1,99 @@
+version: '2'
+services:
+    testdb:
+        image: mariadb:10
+        environment:
+            MYSQL_ROOT_PASSWORD: r00tp@ss!
+            MYSQL_DATABASE: test
+            MYSQL_USER: idbroker
+            MYSQL_PASSWORD: idbroker
+
+    app:
+        build: .
+        depends_on:
+            - testdb
+            - mfaapi
+            - u2fsim
+        working_dir: /data
+        environment:
+            APP_ENV: test
+            EMAIL_SERVICE_accessToken: fake-abc-123
+            EMAIL_SERVICE_assertValidIp: "false"
+            EMAIL_SERVICE_baseUrl: http://email
+            EMAIL_SERVICE_validIpRanges: 192.168.0.0/16
+            EMAIL_SIGNATURE: Dummy Signature for Automated Tests
+            EMAILER_CLASS: \Sil\SilIdBroker\Behat\Context\fakes\FakeEmailer
+            HELP_CENTER_URL: https://help-center
+            IDP_NAME: Test
+            MYSQL_HOST: testdb
+            MYSQL_DATABASE: test
+            MYSQL_USER: idbroker
+            MYSQL_PASSWORD: idbroker
+            API_ACCESS_KEYS: abc123
+            TEST_SERVER_HOSTNAME: localhost
+            PASSWORD_PROFILE_URL: https://www.example.com
+            SUPPORT_EMAIL: support@example.com
+            HR_NOTIFICATIONS_EMAIL: hr@example.com
+            ABANDONED_USER_bestPracticeUrl: http://www.example.com/best-practices.html
+            ABANDONED_USER_deactivateInstructionsUrl: http://www.example.com/deactivate-instructions.html
+            MFA_WEBAUTHN_apiBaseUrl: mfaapi:8080/
+            MFA_WEBAUTHN_apiKey: 10345678-1234-1234-1234-123456789012
+            MFA_WEBAUTHN_apiSecret: 11345678-1234-1234-1234-12345678
+            MFA_WEBAUTHN_appId: ourApp99
+            MFA_WEBAUTHN_rpDisplayName: Our App
+            MFA_WEBAUTHN_rpId: http://app99
+            U2F_SIM_HOST_AND_PORT: u2fsim:8080
+
+    dynamo:
+        image: amazon/dynamodb-local
+        ports:
+            - "8000:8000"
+        environment:
+            AWS_ACCESS_KEY_ID: abc123
+            AWS_SECRET_ACCESS_KEY: abc123
+            AWS_DEFAULT_REGION: us-east-1
+        command: "-jar DynamoDBLocal.jar -sharedDb"
+
+    # Don't use underscores in service names
+    mfaapi:
+        build: ./serverless-mfa-api
+        ports:
+            - 8080
+        environment:
+            AWS_ENDPOINT: dynamo:8000
```

Comment:
I think it would be worth using [extends](https://docs.docker.com/compose/multiple-compose-files/extends/#extending-services-within-the-same-file) and have a base service that has all the common env variables?

Something like
```
services:
  commonenv:
    environment:
      AWS_ACCESS_KEY_ID: abc123
      AWS_SECRET_ACCESS_KEY: abc123
      ...

  mfaapi:
    build: ./serverless-mfa-api
    extends: commonenv
    environment:
      NON_COMMON_VAR: foo
```

OR update file version to 3 and use [extension fields](https://docs.docker.com/compose/compose-file/11-extension/) (like auth-proxy).

Reply:
Yes, but the idea is to get the conversion done quickly and not go for optimization. Thanks for the suggestion.